File: src/doob/data/paths.py
"""Path resolution helpers for the warehouse data lake."""

from __future__ import annotations

from pathlib import Path

from doob.config import warehouse_root


def parquet_path_for_symbol(symbol: str, warehouse: Path | None = None) -> Path:
    """Return the canonical bronze parquet path for an equity ticker."""
    root = warehouse or warehouse_root()
    return (
        root / "data-lake" / "bronze" / "asset_class=equity" / f"symbol={symbol}" / "data.parquet"
    )

File: src/data/paths.rs
/// Parquet path resolution helpers for the warehouse data lake.

use std::path::{Path, PathBuf};

use anyhow::Result;

use crate::config::warehouse_root;

/// Return the canonical bronze parquet path for an equity ticker.
pub fn parquet_path_for_symbol(symbol: &str, warehouse: Option<&Path>) -> Result<PathBuf> {
    let root = match warehouse {
        Some(p) => p.to_path_buf(),
        None => warehouse_root()?,
    };
    Ok(root
        .join("data-lake")
        .join("bronze")
        .join("asset_class=equity")
        .join(format!("symbol={symbol}"))
        .join("data.parquet"))
}

#[cfg(test)]
mod tests {
    use super::*;
    use std::path::Path;

    #[test]
    fn test_path_construction() {
        let wh = Path::new("/tmp/warehouse");
        let path = parquet_path_for_symbol("SPY", Some(wh)).unwrap();
        assert_eq!(
            path,
            PathBuf::from(
                "/tmp/warehouse/data-lake/bronze/asset_class=equity/symbol=SPY/data.parquet"
            )
        );
    }
}
